src/libraries/System.Private.CoreLib/src/System/Boolean.cs

```diff
@@ -191,24 +187,14 @@ public int CompareTo(bool value)
 
         // Custom string compares for early application use by config switches, etc
         //
-        internal static bool IsTrueStringIgnoreCase(ReadOnlySpan<char> value)
-        {
-            return value.Length == 4 &&
-                    (value[0] == 't' || value[0] == 'T') &&
-                    (value[1] == 'r' || value[1] == 'R') &&
-                    (value[2] == 'u' || value[2] == 'U') &&
-                    (value[3] == 'e' || value[3] == 'E');
-        }
+        internal static bool IsTrueStringIgnoreCase(ReadOnlySpan<char> value) =>
```

Comment:
contrary to what I said in the original issue, since these two methods are each only used in 2 places, and it's a hot method, and now they're even shorter than before, I suggest ensuring they inline, and if they don't, measuring with aggressiveinlining.

Reply:
As of right now on my machine on x64, both IsTrue/FalseStringIgnoreCase methods get inlined into TryParse.